### ISSUE TITLE
Create barcode widget

### ISSUE THREAD

**Issue by MartijnR** [kobotoolbox/enketo-express#217]
Not sure if feasible to use webcam and parse barcode on client.

**Comment by MartijnR** [comment]
On mobile it would work okay like a photo input. 

Don't like extracting bar code on server upon submissio.

**Comment by MartijnR** [comment]
http://webintents.org/

**Comment by tinok** [comment]
This might be an option: https://github.com/zxing/zxing/wiki/Scanning-From-Web-Pages

**Comment by MartijnR** [comment]
To look into:
- does it work fully offline
- can we provide a user-friendly UI for unsupported OS/browsers, e.g. desktop
- can we provide a similar experience in iOS and Android

**Comment by barrowsglobal** [comment]
I have managed to get this working with a bit of a hack using 3rd party apps. Maybe someone can clean this up and make it more robust.

TODO:
1. Cleanup code
2. Test offline
3. Extend beyond IOS and Android

3rd party app requierements:
1. Zxing barcode scanner on android
2. Pic2shop barcode scanner on ios

If no app is installed the barcode input should behave as normal allowing you to manually type the value.

I created a widget called barcodescanner in the widgets folder, and included that widget in the widgets.js file.
[barcodescanner.js.zip](https://github.com/kobotoolbox/enketo-express/files/505601/barcodescanner.js.zip)

I hope this helps, and good luck.
